docs/struct.rst

```diff
@@ -0,0 +1,265 @@
+.. Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+   other Shroud Project Developers.
+   See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (BSD-3-Clause)
+
+.. _TypesSandC:
+
+Structs and Classes
```

Comment:
Incidentally, does shroud support inheritance, e.g. ``class D : public B {};`` ?

Reply:
You can wrap class D and class B, but you cannot extend class B in Fortran.  Fortran derived types support an EXTENDS keyword but I don't support that.  But that's a good idea.  Python would need to support it too.  I'll add it to the TODO list.